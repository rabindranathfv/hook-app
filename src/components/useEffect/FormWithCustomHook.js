import React from 'react';
import { useForm } from '../../hooks/useForm';

import './effect.css';

export const FormWithCustomHook = () => {
    
    const [formState, handleInputChange] = useForm({
        name: '',
        email: '',
        password: ''
    })

    const { name, email, password } = formState;

    const handleSubmit = (e) => {
        e.preventDefault();
        console.log(formState)
    }

    return (
        <form onSubmit={handleSubmit}>
            <h1> FormWithCustomHook </h1>
            <hr />

            <div className="form-group">
                <label htmlFor=""></label>
                <input type="text"
                className="form-control" name="name" id="" aria-describedby="name-form" placeholder="enter your name" value={name} onChange={handleInputChange}/>
                <small className="form-text text-muted">Name</small>

                <label htmlFor=""></label>
                <input type="email"
                className="form-control" name="email" id="" aria-describedby="email-form" placeholder="enter your email" value={email} onChange={handleInputChange}/>
                <small className="form-text text-muted">Email</small>

                <label htmlFor=""></label>
                <input type="password"
                className="form-control" name="password" id="" aria-describedby="password-form" placeholder="enter your password" value={password} onChange={handleInputChange}/>
                <small className="form-text text-muted">password</small>
            </div>

            <button type="submit" className="btn btn-outline-primary"> Submit</button>
        </form>
    )
}
