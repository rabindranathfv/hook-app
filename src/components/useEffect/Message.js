import React, { useEffect } from 'react';

export const Message = () => {

    useEffect(() => {
        console.log('Message mount');
        return () => {
            console.log('message unmount')
        }
    }, []);

    return (
        <div>
            <h4> you are cool</h4>
        </div>
    )
}
