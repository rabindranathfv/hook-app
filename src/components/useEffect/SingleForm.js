import React, { useState, useEffect } from 'react';
import './effect.css';

export const SingleForm = () => {

    const [formState, setFormState] = useState({
        name: '',
        email: ''
    })

    const { name, email } = formState;

    // just execute one time at render component
    useEffect(() => {
        console.log('hey!!')
    }, []);

    // execute every time form updates or re-render
    useEffect(() => {
        console.log('Listem all formState changes')
    }, [formState])

    useEffect(() => {
        console.log('listem email changes')
    }, [email])

    const handleInputChange = (e) => {
        setFormState({
            ...formState,
            [ e.target.name ] : e.target.value
        })
    };

    return (
        <div>
            <h1> useEffect </h1>
            <hr />

            <div className="form-group">
                <label htmlFor=""></label>
                <input type="text"
                className="form-control" name="name" id="" aria-describedby="name-form" placeholder="enter your name" value={name} onChange={handleInputChange}/>
                <small className="form-text text-muted">Name</small>

                <label htmlFor=""></label>
                <input type="text"
                className="form-control" name="email" id="" aria-describedby="email-form" placeholder="enter your email" value={email} onChange={handleInputChange}/>
                <small className="form-text text-muted">Email</small>
            </div>
        </div>
    )
}
