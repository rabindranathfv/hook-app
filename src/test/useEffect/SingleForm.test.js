import React from 'react';
import { shallow } from 'enzyme';
import { SingleForm } from '../../components/useEffect/SingleForm';

describe('SingleForm::', () => {

    let wrapper = shallow(<SingleForm />);

    beforeEach( ()=> {
        wrapper = shallow(<SingleForm />);
    })
    
    it('should return default values SingleForm', () => {
        expect(wrapper).toMatchSnapshot();
    });

    // it('should render with default values in all counters', () => {
    //     const counterText = wrapper.find('h1').at(0).text().trim();
    //     const counterText2 = wrapper.find('h1').at(1).text().trim();

    //     expect(counterText).toBe('Counter1 0');
    //     expect(counterText2).toBe('Counter2 20');
    // });

    // it('Should increment counter', () => {
    //     wrapper.find('button').at(0).simulate('click');
    //     const counterText = wrapper.find('h1').at(0).text().trim();
        
    //     expect(counterText).toBe('Counter1 1');
    // });

    // it('Should decrement counter', () => {
    //     wrapper.find('button').at(1).simulate('click');
    //     const counterText = wrapper.find('h1').at(0).text().trim();
        
    //     expect(counterText).toBe('Counter1 -1');
    // });

    // it('Should decrement reset', () => {
    //     wrapper.find('button').at(2).simulate('click');
    //     const counterText = wrapper.find('h1').at(0).text().trim();
        
    //     expect(counterText).toBe('Counter1 0');
    // });
})
