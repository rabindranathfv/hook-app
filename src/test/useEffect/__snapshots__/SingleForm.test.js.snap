// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SingleForm:: should return default values SingleForm 1`] = `
<div>
  <h1>
     useEffect 
  </h1>
  <hr />
  <div
    className="form-group"
  >
    <label
      htmlFor=""
    />
    <input
      aria-describedby="name-form"
      className="form-control"
      id=""
      name="name"
      onChange={[Function]}
      placeholder="enter your name"
      type="text"
      value=""
    />
    <small
      className="form-text text-muted"
    >
      Name
    </small>
    <label
      htmlFor=""
    />
    <input
      aria-describedby="email-form"
      className="form-control"
      id=""
      name="email"
      onChange={[Function]}
      placeholder="enter your email"
      type="text"
      value=""
    />
    <small
      className="form-text text-muted"
    >
      Email
    </small>
  </div>
</div>
`;
